test/test_rpc_endpoints.py

```diff
@@ -0,0 +1,641 @@
+
+from .util import (
+    load_file_content,
+)
+from .test_endpoints import send_transaction
+
+
+DEPLOY_CONTENT = load_file_content("deploy.json")
+INVOKE_CONTENT = load_file_content("invoke.json")
+
+
+def rpc_call(method: str, params: dict | list) -> dict:
+    """
+    Make a call to the RPC endpoint
+    """
+    req = {
+        "jsonrpc": "2.0",
+        "method": method,
+        "params": params,
+        "id": 0
+    }
+
+    resp = app.test_client().post(
+        "/rpc",
+        content_type="application/json",
+        data=json.dumps(req)
+    )
+    result = json.loads(resp.data.decode("utf-8"))
+    return result
+
+
+def gateway_call(method: str, **kwargs):
+    """
+    Make a call to the gateway
+    """
+    resp = app.test_client().get(
+        f"/feeder_gateway/{method}?{''.join(f'{key}={value}&' for key, value in kwargs.items()).rstrip('&')}"
+    )
+    return json.loads(resp.data.decode("utf-8"))
+
+
+@pytest.fixture(name="deploy_info")
+def fixture_deploy_info() -> dict:
+    """
+    Deploy a contract on devnet and return deployment info dict
+    """
+    resp = send_transaction(json.loads(DEPLOY_CONTENT))
+    deploy_info = json.loads(resp.data.decode("utf-8"))
+    return deploy_info
+
+
+@pytest.fixture(name="invoke_info")
+def fixture_invoke_info() -> dict:
+    """
+    Make a invoke transaction on devnet and return invoke info dict
+    """
+    invoke_tx = json.loads(INVOKE_CONTENT)
+    invoke_tx["calldata"] = ["0"]
+    resp = send_transaction(invoke_tx)
+    invoke_info = json.loads(resp.data.decode("utf-8"))
+    return invoke_info
+
+
+def pad_zero(felt: str) -> str:
+    """
+    Convert felt with format `0xValue` to format `0x0Value`
+    """
+    felt = felt.lstrip("0x")
+    return "0x0" + felt
+
+
+# pylint: disable=unused-argument
+def test_get_block_by_number(deploy_info):
+    """
+    Get block by number
+    """
+    resp = rpc_call(
+        "starknet_getBlockByNumber", params={"block_number": 0}
+    )
+    block = resp["result"]
+    expected = {
+        "parent_hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "block_number": 0,
+        "status": "ACCEPTED_ON_L2",
+        "sequencer": "0x0000000000000000000000000000000000000000",
+        "old_root": "0x0",
+    }
+
+    for key, value in expected.items():
+        assert block[key] == value
+
+
+def test_get_block_by_number_raises_on_incorrect_number(deploy_info):
+    """
+    Get block by incorrect number
+    """
+    ex = rpc_call(
+        "starknet_getBlockByNumber", params={"block_number": 1234}
+    )
+
+    assert ex["error"] == {
+        "code": 26,
+        "message": "Invalid block number"
+    }
+
+
+def test_get_block_by_hash(deploy_info):
+    """
+    Get block by hash
+    """
+    block_hash = gateway_call("get_block", blockNumber=0)["block_hash"]
+
+    resp = rpc_call(
+        "starknet_getBlockByHash", params={"block_hash": block_hash}
+    )
+    block2 = resp["result"]
```

Comment:
Why block2?

Reply:
My mistake, leftover from the previous version of this test.